app/controllers/posts_controller.rb

```diff
@@ -8,5 +9,26 @@ def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
-  # 1
+
+  private
+
+  def create
+    new_post = current_user.posts.new(post_params)
+    new_post.likes_counter = 0
+    new_post.comments_counter = 0
+    new_post.update_posts_counter
+    respond_to do |format|
+      format.html do
+        if new_post.save
+          redirect_to "/users/#{new_post.user.id}/posts/", notice: 'Success!'
+        else
+          render :new, alert: 'Error!'
```

Comment:
- [ ] I am not able to create a post as it's giving an error that it can't able to find the `create` method. Please fix it. Also, use `new` method to create an instance just like you did in `Comments controller` also include a `new` route for the posts in `config/routes.rb` file.

![Screenshot from 2022-04-01 23-09-32](https://user-images.githubusercontent.com/22434660/161314706-fc0483aa-6d27-4450-8567-904d5a54eb59.png)

Reply:
Thank you @Dipeshtwis  i have implemented the requested changes